cmd/vclusterctl/cmd/list.go

```diff
@@ -143,3 +147,172 @@ func (cmd *ListCmd) Run(cobraCmd *cobra.Command, args []string) error {
 
 	return nil
 }
+
+func getVCluster(object interface{}, v string, client *kubernetes.Clientset, kubeClientConfig clientcmd.ClientConfig) (VCluster, error) {
+	var status string
+	var namespace string
+	var created time.Time
+	switch v := object.(type) {
+	case appsv1.Deployment:
+	case appsv1.StatefulSet:
+		namespace = v.Namespace
+		created = v.GetCreationTimestamp().Time
```

Comment:
If you are using v.Namespace and v.Annotations, just use v.CreationTimestamp as well